### PR TITLE
DLPX-63519 ATA reads timeout after Azure update

### PR DESCRIPTION
**Pull request by danmcd**
Reviewed at: http://reviews.delphix.com/r/48492/

This is Pavel's commit I'm shepherding into illumos-omnios.  In hyperv.c, I changed his printf() to cmn_err() to match existing code.  I plan on testing this on Azure itself later today if all goes well.

### CODE REVIEW

**Comment by danmcd** [comment]
Also, I'd highly recommend this get cherry-picked to 030, as I'm going to be testing on '030 + this patch.

File: usr/src/uts/intel/io/dktp/controller/ata/ata_common.c [hunk 1]
```diff
@@ -180,6 +182,11 @@ char *ata_dev_DMA_sel_msg;
 static	struct bus_ops	 ata_bus_ops;
 static	struct bus_ops	*scsa_bus_ops_p;
 
+/*
+ * ATA workaround for Azure bug
+ */
+int ata_azure_workaround = FALSE;
```
Comment (citrus-it) [inline usr/src/uts/intel/io/dktp/controller/ata/ata_common.c]: Can you make this a boolean?

Comment (danmcd) [reply usr/src/uts/intel/io/dktp/controller/ata/ata_common.c]: Done.

**Comment by danmcd** [comment]
NOTE: Tested on Azure with an r151030 build.  (Thanks @hadfl for the VHD build!)

**Pull request merged**